app/controllers/users.js

```diff
@@ -36,153 +49,199 @@ exports.signup = function(req, res) {
 };
 
-/** 
+/**
  * Check avatar - Confirm if the user who logged in via passport
  * already has an avatar. If they don't have one, redirect them
  * to our Choose an Avatar page.
+ * @param {any} req - the request object
+ * @param {any} res - the response object
+ * @returns {func} Assigns an avatar to a user
  */
-exports.checkAvatar = function(req, res) {
+exports.checkAvatar = (req, res) => {
   if (req.user && req.user._id) {
     User.findOne({
       _id: req.user._id
     })
-    .exec(function(err, user) {
-      if (user.avatar !== undefined) {
-        res.redirect('/#!/');
-      } else {
-        res.redirect('/#!/choose-avatar');
-      }
-    });
+      .exec((err, user) => {
+        if (user.avatar !== undefined) {
+          res.redirect('/#!/');
+        } else {
+          res.redirect('/#!/choose-avatar');
+        }
+      });
   } else {
     // If user doesn't even exist, redirect to /
     res.redirect('/');
   }
-
 };
 
 /**
  * Create user
+ * Function that creates a user and assigns a token to them
+ * @param {any} req - the request object
+ * @param {any} res - the response object
+ * @returns {func} The created user and their assigned token
  */
-exports.create = function(req, res) {
-  if (req.body.name && req.body.password && req.body.email) {
-    User.findOne({
-      email: req.body.email
-    }).exec(function(err,existingUser) {
-      if (!existingUser) {
-        var user = new User(req.body);
-        // Switch the user's avatar index to an actual avatar url
-        user.avatar = avatars[user.avatar];
-        user.provider = 'local';
-        user.save(function(err) {
-          if (err) {
-            return res.render('/#!/signup?error=unknown', {
-              errors: err.errors,
-              user: user
-            });
-          }
-          req.logIn(user, function(err) {
-            if (err) return next(err);
-            return res.redirect('/#!/');
-          });
+exports.create = (req, res) => {
+  const { errors, valid } = validateSignup(req.body);
+  if (!valid) {
+    return res.status(400).json(errors);
+  }
+
+  User.findOne({
+    email: req.body.email,
+  }).then((foundUser) => {
+    if (foundUser) {
+      return res.redirect('/#!/signup?error=existinguser');
+    }
+    const user = new User(req.body);
+    user.avatar = avatars[user.avatar];
+    user.provider = 'local';
+    user.save((err) => {
+      if (err) {
+        return res.status(500).json({
+          status: 'Error',
+          message: err.errors,
         });
-      } else {
-        return res.redirect('/#!/signup?error=existinguser');
       }
+      req.logIn(user, (err, next) => {
+        if (err) return next(err);
+        const token = jwt.sign(
+          {
+            id: user._id,
+          },
+          'somesecretkeyhere', { expiresIn: '30 days' }
+        );
+        return res.status(201).json({
+          status: 'Success',
+          message: 'Registration Successful',
+          data: {
```

Comment:
`data` is a pretty ambiguous variable name. Can we use another variable name apart from `data`?

Reply:
Also, why are we only sending back the token? won't we need some properties of the user object in the app?